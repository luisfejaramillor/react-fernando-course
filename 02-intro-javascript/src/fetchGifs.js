import { randomUrl } from "./constants/apiKey"

export const getRandomGif = async ()=> {
    try {
        const result = await fetch(`${randomUrl}`)

        if(!result.ok){
            throw new Error(`Failed to fetch: ${result.status}`);
        }

        const obj = await result.json();
        console.log(obj)

    } catch (error) {
        console.error('Error fetching BREAKING BAD QUOTES:', error);
    }


}