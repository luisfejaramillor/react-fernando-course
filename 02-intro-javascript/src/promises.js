// import { getHeroById } from "./getHeroById";

// export const getHeroByIdAsync = (id) => {
//     return(
//          new Promise((resolve, reject) => {
//             setTimeout(() => {
//                 const hero = getHeroById(id);
//                 hero ? resolve(hero) : reject(`Could not found hero with id: ${id}`)
//             }, 2000);
//         })
//     )
    
// }
