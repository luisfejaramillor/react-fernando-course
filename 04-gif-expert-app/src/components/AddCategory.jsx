import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputValue = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    
    if (inputValue.trim() === "") return;
    setInputValue("");
    onNewCategory(inputValue.trim());
  };
  return (
    <form type="text" onSubmit={handleSubmit} aria-label="form">
      <input type="text" onChange={handleInputValue} value={inputValue} />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
