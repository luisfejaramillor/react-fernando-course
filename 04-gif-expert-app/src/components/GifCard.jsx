import PropTypes from "prop-types";



export const GifCard = ({title, img}) => {
  return (
    <div className="card" >
        <img src={img} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifCard.propTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
}
