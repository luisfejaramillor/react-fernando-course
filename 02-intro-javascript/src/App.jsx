import './App.css'
import { getRandomGif } from './fetchGifs'


getRandomGif()

function App() {

  return (
    <>
      <h1>HOli</h1>
    </>
  )
}

export default App
