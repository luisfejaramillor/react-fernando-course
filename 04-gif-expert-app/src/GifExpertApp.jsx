import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Breaking bad"]);

  const handleNewCategory = (category) => {
    if (categories.includes(category)) return;
    setCategories([category, ...categories]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={handleNewCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
