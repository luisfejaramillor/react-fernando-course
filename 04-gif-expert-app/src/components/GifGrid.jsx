import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifCard } from "./GifCard";

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <div>
      <h3> {category} </h3>
      {isLoading && <h2>Loading...</h2>}
      <div className="card-grid">
        {gifs.map((gif) => (
          <GifCard key={gif.id} {...gif} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
