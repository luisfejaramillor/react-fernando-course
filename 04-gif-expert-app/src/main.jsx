import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifExpertApp'

import './styles/main.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <GifExpertApp />
)
