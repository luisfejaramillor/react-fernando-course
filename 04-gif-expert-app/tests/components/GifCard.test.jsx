import { render, screen } from "@testing-library/react";
import { GifCard } from "../../src/components";
describe("Testing GifCard component", () => {
  const title = "Test title";
  const img = "https://one-punch.com/saitama.jpg";
  test("Testing sobre el snapshot", () => {
    const { container } = render(<GifCard title={title} img={img} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el url y alt indicado", () => {
    render(<GifCard title={title} img={img} />);
    const { alt, src } = screen.getByRole("img");
    expect(alt).toBe(title);
    expect(src).toBe(img);
  });

  test("should show title in the component", () => {
    render(<GifCard title={title} img={img} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
