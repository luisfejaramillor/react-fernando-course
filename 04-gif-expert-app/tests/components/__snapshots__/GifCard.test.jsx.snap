// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifCard component Testing sobre el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
